pkg/gameserverallocations/controller.go

```diff
@@ -146,16 +146,32 @@ func NewController(apiServer *apiserver.APIServer,
 			// only interested in if the old / new state was/is Ready
 			oldGs := oldObj.(*stablev1alpha1.GameServer)
 			newGs := newObj.(*stablev1alpha1.GameServer)
-			if oldGs.Status.State == stablev1alpha1.GameServerStateReady || newGs.Status.State == stablev1alpha1.GameServerStateReady {
-				if key, ok := c.getKey(newGs); ok {
-					if newGs.Status.State == stablev1alpha1.GameServerStateReady {
-						c.readyGameServers.Store(key, newGs)
-					} else {
-						c.readyGameServers.Delete(key)
-					}
+			key, ok := c.getKey(newGs)
+			if !ok {
+				return
+			}
+			if newGs.IsBeingDeleted() {
+				// TOXO: test this
```

Comment:
Comments probably redundant.
Why TOXO, not TODO? Is it an abbreviation or just to distinguish from TODO?